pkg/metrics/metrics.jsx

```diff
@@ -507,62 +530,140 @@
+                // user or root user if the user is privileged. Other users
+                // containers will show up under the user@uid cgroup
+                const matches = name.match(/libpod-(?<containerid>[a-z|0-9]{64})\.scope/);
+                if (matches && valuesA[i] !== undefined && valuesB[i] !== undefined) {
+                    const containerid = matches.groups.containerid;
+                    const umatches = name.match(useridCgroupRe);
+                    const isUser = !!umatches;
+                    const uid = parseInt(umatches?.groups.userid) || 0;
+
+                    if (uid === 0 || this.state.userid == uid) {
+                        merged.push([containerid, valuesA[i], valuesB[i], isUser, true, uid]);
+                        return;
+                    }
+                }
+
+                // combine user slices into user@ID
+                // { name: [ read, write ] }
+                const umatches = name.match(useridCgroupRe);
+                if (umatches) {
+                    if (userSlices[umatches.groups.userid] === undefined) {
+                        userSlices[umatches.groups.userid] = [valuesA[i], valuesB[i]];
+                    } else {
+                        userSlices[umatches.groups.userid][0] += valuesA[i];
+                        userSlices[umatches.groups.userid][1] += valuesB[i];
+                    }
+                }
+            });
+
+            Object.keys(userSlices).forEach((key) => {
+                merged.push(["user@" + key, userSlices[key][0], userSlices[key][1], false, false]);
+            });
+
+            // sort by overall (read + write) disk usage
+            merged.sort((a, b) => (b[1] + b[2]) - (a[1] + a[2]));
+            return merged.slice(0, n);
+        };
+
+        const newState = {};
+
+        const topServicesDiskIO = n_biggest(5, this.cgroupDiskNames, this.privilegedSamples[0], this.privilegedSamples[1]);
+        newState.topServicesDiskIO = topServicesDiskIO.filter(([_, read, write, ..._rest]) => read !== 0 || write !== 0).map(([name, read, write, isUser, isContainer, uid]) => {
+            return this.cgroupRow(name, isUser, isContainer, uid, read > 1 ? cockpit.format_bytes_per_sec(read) : 0, write > 1 ? cockpit.format_bytes_per_sec(write) : 0);
+        });
+
+        this.setState(newState);
+    }
+
+    getCachedPodName = (uid, containerid) => this.state.podNameMapping[uid] && this.state.podNameMapping[uid][containerid];
+
+    cgroupRow = (name, is_user, is_container, uid, ...values) => {
+        const podman_installed = cockpit.manifests && cockpit.manifests.podman;
+
+        const cgroupClickHandler = (name, isUser, isContainer, uid) => {
+            if (isContainer) {
+                const containerName = this.getCachedPodName(uid, name);
+                if (containerName) {
+                    cockpit.jump("/podman#/?name=" + containerName);
+                } else {
+                    cockpit.jump("/podman");
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-17692-20230504-071225-f305d649-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)